File: dotnet-core/Tea/Decorators/Drivers/BrowserBase.cs
using System.Collections.Generic;
using OpenQA.Selenium;

namespace Tea
{
    public enum BrowserList
    {
        Chrome,
        Firefox,
        IE,
        Edge,
        Opera
    }

    public abstract class BrowserBase
    {
        public abstract void Start(BrowserList browser);
        public abstract void Quit();
        public abstract void GoTo(string url);
        public abstract Element FindElement(By locator);
        public abstract List<Element> FindElements(By locator);
        public abstract Element FindElementInFrames(By by);
        public abstract List<Element> FindAllFrames();
        public abstract Element GetActiveElement();
        public abstract IWebDriver GetDriver();
    }
}

File: java/tea-framework/src/main/java/com/tea/framework/SeleniumWrapper/Drivers/BrowserBase.java
package com.tea.framework.SeleniumWrapper.Drivers;

import com.tea.framework.SeleniumWrapper.Elements.Element;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import java.util.List;

public interface BrowserBase {
    void Start(BrowserList browser);

    void Quit();

    void GoTo(String url);

    Element FindElement(By locator);

    List<Element> FindElements(By locator);

    Element FindElementInFrames(By by);

    List<Element> FindAllFrames();

    Element GetActiveElement();

    WebDriver GetDriver();
}


